src/parser.rs

```diff
@@ -1683,6 +1689,19 @@ impl Parser {
         })
     }
 
+    fn parse_show_objects(&mut self) -> Option<Statement> {
+        let object_type = self
+            .parse_one_of_keywords(&["SOURCES", "VIEWS", "SINKS", "TABLES"])
+            .map(|s| match s {
+                "SOURCES" => ObjectType::Source,
+                "VIEWS" => ObjectType::View,
+                "SINKS" => ObjectType::Sink,
+                "TABLES" => ObjectType::Table,
+                _ => unimplemented!(),
```

Comment:
`unimplemented!` panics (it will crash the program/is not a recoverable error) so we shouldn't use it to handle user input.

Reply:
It is not really handling user input in the sense you mean. `parse_one_of_keywords(&["foo", "bar"])` cannot possibly return something other than `Some("foo")`, `Some("bar")`, or `None`.

Reply:
ah, that makes sense, in that case I would prefer to put in an explicit `val => panic!("parse_one_of_keywords returned an impossible value: {}", val),